src/imageEditor.re

```diff
@@ -0,0 +1,100 @@
+type imageSource = [ | `URI(string) | `Required(Packager.required)];
+
+type offset = {
+  x: int,
+  y: int,
+};
+
+type size = {
+  width: int,
+  height: int,
+};
+
+[@bs.deriving jsConverter]
+type resizeModeT = [
+  | [@bs.as "contain"] `Contain
+  | [@bs.as "cover"] `Cover
+  | [@bs.as "stretch"] `Stretch
```

Comment:
All those bs.as are unnecessary

```
Warning number 101
  /home/opam/bs-react-native/src/imageEditor.rei 15:7-11
  
  13 │ [@bs.deriving jsConverter]
  14 │ type resizeModeT = [
  15 │   | [@bs.as "contain"] `Contain
  16 │   | [@bs.as "cover"] `Cover
  17 │   | [@bs.as "stretch"] `Stretch
  
  Unused BuckleScript attribute: bs.as

  Warning number 101
  /home/opam/bs-react-native/src/imageEditor.rei 16:7-11
  
  14 │ type resizeModeT = [
  15 │   | [@bs.as "contain"] `Contain
  16 │   | [@bs.as "cover"] `Cover
  17 │   | [@bs.as "stretch"] `Stretch
  18 │ ];
  
  Unused BuckleScript attribute: bs.as

  Warning number 101
  /home/opam/bs-react-native/src/imageEditor.rei 17:7-11
  
  15 │   | [@bs.as "contain"] `Contain
  16 │   | [@bs.as "cover"] `Cover
  17 │   | [@bs.as "stretch"] `Stretch
  18 │ ];
  19 │ 
  
  Unused BuckleScript attribute: bs.as
```

Reply:
I guess bs automatically transform those according to the name of the variant?!

Reply:
Without `bs.as`, the first letter will be uppercase which will cause an error with RN.

In the interface, however, there is no need to expose the `jsConverter` function, so only the variants should suffice (`imageEditor.rei` copied below, excluding documentation):

```
type imageSource = [ | `URI(string) | `Required(Packager.required)];

type offset = {
  x: int,
  y: int,
};

type size = {
  width: int,
  height: int,
};

type resizeModeT = [ | `Contain | `Cover | `Stretch ];

[@bs.deriving abstract]
type error = pri {
  code: int,
  message: string,
};

let cropImage:
  (
    ~uri: imageSource,
    ~offset: offset,
    ~size: size,
    ~displaySize: size=?,
    ~resizeMode: resizeModeT=?,
    ~success: string => unit,
    ~failure: error => unit,
    unit
  ) =>
  unit;
```

Reply:
Oh right, I didn't read correctly the filename in the error. My bad.